bigchaindb/core.py

```diff
@@ -190,6 +155,40 @@ def get_stale_transactions(self):
             .filter(lambda tx: time() - tx['assignment_timestamp'] >
                     self.backlog_reassign_delay).run(self.conn)
 
+    def validate_transaction(self, transaction):
+        """Validate a transaction.
+
+        Args:
+            transaction (Transaction): transaction to validate.
+
+        Returns:
+            The transaction if the transaction is valid else it raises an
+            exception describing the reason why the transaction is invalid.
+        """
+
+        return self.consensus.validate_transaction(self, transaction)
+
+    def is_valid_transaction(self, transaction):
+        """Check whether a transaction is valid or invalid.
+
+        Similar to `validate_transaction` but never raises an exception.
+        It returns `False` if the transaction is invalid.
+
+        Args:
+            transaction (Transaction): transaction to check.
+
+        Returns:
+            `transaction` (Transaction) if the transaction is valid, otherwise
+            `False`
+        """
+
+        try:
+            return self.validate_transaction(transaction)
+        except (ValueError, exceptions.OperationError, exceptions.TransactionDoesNotExist,
+                exceptions.TransactionOwnerError, exceptions.DoubleSpend,
+                exceptions.InvalidHash, exceptions.InvalidSignature):
+            return False
```

Comment:
Oh sorry, I meant to convey the idea that if a certain group of exceptions share certain characteristics, and for this reason are often caught together (as above), then it may make sense to have a parent class from which these classes inherit. One advantage of doing so would be that in some cases the parent class can be caught if all children are involved.

Example:

```python
        try:
            return self.validate_transaction(transaction)
        except (ValueError, exceptions.OperationError, exceptions.TransactionError):
            return False
```